scrapi/base/transformer.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import unicode_literals
+
+import abc
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTransformer(object):
+
+    __metaclass__ = abc.ABCMeta
+
+    def transform(self, doc):
+        return self._transform(self.schema, doc)
+
+    def _transform(self, schema, doc):
+        transformed = {}
```

Comment:
_transform takes a schema which it can recurse on, you can't recurse on self.schema